### PR TITLE
#436 | log list virtualization

### PR DESCRIPTION
**Pull request by clmrv**
## Description
Add log list virtualization
Also sets up following logs (to be improved)

## Motivation and Context
#436

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](https://github.com/wttech/cogboard/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] Automated functional tests have been added or modified to cover my changes (if applicable)
- [ ] I have updated the documentation accordingly.

---

I hereby agree to the terms of the Cogboard Contributor License Agreement.

### CODE REVIEW

File: cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js [hunk 1]
```diff
@@ -43,6 +51,26 @@ export default function LogList({
     </VariableGridSchema>
   );
 
+  useEffect(() => {
+    if (shouldFollowLogs) {
+      console.log(scrollerRef.current.scrollHeight);
+      console.log(scrollerRef.current.scrollTop);
```
Comment (janeksmielowski) [inline cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js:57]: wywalmy console logi

File: cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js [hunk 2]
```diff
@@ -43,6 +51,26 @@ export default function LogList({
...
+      scrollerRef.current.scrollTo({
+        top: scrollerRef.current.scrollHeight,
+        behavior: 'smooth'
+      });
+    }
+    setScroll(scrollerRef.current.scrollTop);
+    console.log(scroll);
```
Comment (janeksmielowski) [inline cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js:64]: j/w

File: cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js [hunk 3]
```diff
@@ -54,18 +82,27 @@ export default function LogList({
...
-            search={search}
-            highlight={isLogHighlighted(log, search)}
-          />
-        ))}
+        <StyledVirtuoso
+          scrollerRef={ref => (scrollerRef.current = ref)}
+          isScrolling={isScrolling => isScrolling && stopFollowingOnUpScroll()}
+          totalCount={filteredLogs.length}
+          increaseViewportBy={300} // defines loading overlap (in pixels)
+          itemContent={index => {
```
Comment (janeksmielowski) [inline cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js:90]: wynieśmy to całe do funkcji

File: cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/styled.js [hunk 4]
```diff
@@ -76,7 +80,7 @@ export const CustomAccordion = styled(Accordion)`
     overflow: hidden;
   }
   &.Mui-expanded {
-    margin: 0.5em 0;
+    margin: 0.5em 0 !important;
```
Comment (janeksmielowski) [inline cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/styled.js:83]: wolałbym nie zaszywać !importantów w kodzie. Spróbujmy zamiast tego skonstruować silniejszy selektor. Na przykład podwójny ampersand `&&.Mui-expanded`

**Review by janeksmielowski — CHANGES_REQUESTED** [review]

File: cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/styled.js (hunk 4 above)
Comment (clmrv) [reply cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/styled.js]: Fajny trik z tym `&&`

Comment (janeksmielowski) [reply cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/styled.js]: 😎

**Review by janeksmielowski — APPROVED** [review]

**Pull request merged**